File: DynamicProgramming/26jumpGame/jumpGame.py
# leetcode 55
class Solution:
    def jumpGame(self, nums: list[int]) -> bool:
        goal = len(nums) - 1
        for i in range(len(nums) - 1, -1, -1):
            if i + nums[i] >= goal:
                goal = i
        return True if goal == 0 else False
print(Solution().jumpGame([2,3,1,1,4]))
print(Solution().jumpGame([3,2,1,0,4]))

File: DynamicProgramming/26jumpGame/jumpGame.java
public class jumpGame {
    public static boolean jump(int[] nums) {
        int goal = nums.length - 1;
        for (int i = nums.length - 1; i > -1; i--) {
            if ((i + nums[i]) >= goal) {
                goal = i;
            }
        }
        return goal == 0 ? true : false;
    }

    public static void main(String[] args) {
        System.out.println(jump(new int[] { 2, 3, 1, 1, 4 }));
        System.out.println(jump(new int[] { 3, 2, 1, 0, 4 }));
    }
}